src/odemis/acq/fastem.py

```diff
@@ -281,10 +281,12 @@ def __init__(self, scanner, multibeam, descanner, detector, stage, scan_stage, c
         # save the initial multibeam resolution, because the resolution will get updated if save_full_cells is True
         self._old_res = self._multibeam.resolution.value
 
-        path = fastem_util.create_image_dir("beam-shift-correction")
-        # The path should be [image-dir]/beam-shift-correction/[timestamp]/[roa-name]
-        self.path = os.path.join(path, self._roa.name.value)
-        os.makedirs(self.path, exist_ok=True)
+        beam_shift_path = fastem_util.create_image_dir("beam-shift-correction")
+        # The path should be [image-dir]/beam-shift-correction/[timestamp]/[project-name]/[roa-name]_[slice-idx]
```

Comment:
it seems that the `path` argument can be anything, so the [timestamp]/[project-name] is not a given. If it should always be of that format, maybe introduce a custom datatype and loosely enforce it via typing.

Reply:
It really should only be a string or None, so for now I went for the simpler fix: `path if path else "",`. Let me know if you do not agree with that.